interview-questions.md

```diff
@@ -7,64 +7,64 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-  Your answer:
+  Your answer: A block of reusable code that can be called as many times as desired to do some sort of action. You use a function to avoid repeating the same lines of code that can be done simply by calling the function. It also saves you a lot of time.
 
-  Researched answer:
+  Researched answer: A function is an important feature of Javascript in which a set of statements do some type of task such as calculating the sum of two numbers. It takes an input and returns and output. A function consists of the name of the function, a list of parameters inside parentheses and separated by commas, and a statement that defines what the function does inside a set of curly brackets. Primitive data types can be passed to functions in place of a parameter. However, this change to the parameter does not change globally (i.e. outside the scope of the function) or to the called function. Non-primitive data types such as arrays or objects can also be passed as a parameter. Functions do not require a name and are considered to be anonymous. However, providing a name allows the user to understand what the function itself does, making it easier to identify when debugging.
 
 
 
 2. What is the difference between .map() and .filter()?
 
-  Your answer:
+  Your answer: The function .map() goes through and modifies the entire array at all indices and the result is an array of the same length as the original array. The .filter() function only keeps the values (or modified values) in the array that satisfies a certain condition, so ultimately the array returned is shorter than the original array.
 
-  Researched answer:
+  Researched answer: The .map() method creates a new array that is made up results of calling a given function on every item in the calling array. It provides a callback function once for each element in an array in order. With that information, it makes a new array from those results. It can accept three agruments: value, index, and the array being mapped. The .filter() method creates a new array with the items that pass the condition implemented by the function. It also provides a callback function once for each element in an array BUT the only difference from .map() is that it only returns values that are true to the given condition. Elements that do no pass the callback are skipped and not included in the new array. It also takes the same three arguments as .map(). It does not mutate the array in which it is called, making it very different from the orignal array.
 
 
 
 3. What is the difference between console.log() and return?
 
-  Your answer:
+  Your answer: The console.log() is what lines of code the user wants to appear in the machine or in the console either in the terminal or the console in a browser. It can be useful for finding bugs or to see changes in the code. A return is the end value of some logic like a for loop or an if statement. There is only one return per loop or function.
```

Comment:
Absolutely. Another way we can describe a return is what the output is to the actions of our code. A console log is used for debugging or inspecting outcomes/sections of our data being processed, while the return is the actual finished product of running our code.